dsp/transform/hilbert_test.go

```diff
@@ -0,0 +1,52 @@
+// Copyright ©2024 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package transform
+
+import (
+	"testing"
+
+	"gonum.org/v1/gonum/cmplxs"
+)
+
+var hilbertAnalyticSignalTests = []struct {
+	name string
+	in   []float64
+	want []complex128
+}{
+	{"Hilbert nil slice", nil, nil},
+	{"Hilbert empty slice", []float64{}, []complex128{}},
+	{"Hilbert zeros test", []float64{0, 0, 0, 0}, []complex128{0, 0, 0, 0}},
+	{"Hilbert whole components test", []float64{1, 2, 3, 4}, []complex128{1 + 1i, 2 - 1i, 3 - 1i, 4 + 1i}},
+	{
```

Comment:
Usually, I do a map[string]struct

Where the string is the name, it avoids adding it to the struct and make sure to provide unique name in case of copypasta

Reply:
> I do a map[string]struct

Please don't do this here.

Reply:
OK.

I'm fine with having no map[string]struct but do you have any reason you could share

Reply:
Consistent test ordering.

Reply:
I have never had a need for that, except an `//Output:` in an `ExampleFoo`

But OK, I'm fine with this